script.js

```diff
@@ -0,0 +1,201 @@
+
+  //textContent로 불러올 시 "  \n"이 더해지는 현상이 있어 slice로 처리
+
+  doneList.removeChild(doneItem);
+  editLocalStorage(id, { text: itemText, type: "todo" });
+
+  createTodoItem(id, itemText);
+  updateCnt();
+
+  // button 기준 parentELement = todoItem
+  // Unique ID를 기준으로 localStorage 수정
+}
+
+// ====== Item 조작 관련 함수들 ======= //
+
+function deleteTodoItem(e) {
+  const todoItem = e.currentTarget.parentElement;
+  const id = todoItem.getAttribute("id");
+  todoList.removeChild(todoItem);
+  removeFromLocalStorage(id);
+  updateCnt();
+
+  // button 기준 parentELement = doneItem
+}
+
+function deleteDoneItem(e) {
+  const doneItem = e.currentTarget.parentElement;
+  const id = doneItem.getAttribute("id");
+  doneList.removeChild(doneItem);
+  removeFromLocalStorage(id);
+  updateCnt();
+
+  // button 기준 parentELement = todoItem
+}
+
+function createTodoItem(id, itemText) {
+  const todoItem = document.createElement("div");
+  let attribute = document.createAttribute("id");
+  attribute.value = id;
+  todoItem.setAttributeNode(attribute);
+  todoItem.classList.add("item");
+
+  // 새로운 div element 생성 후 인자로 전달된 unique id를 attribute으로 설정
+  // 'item' 클래스는 각 아이템의 css 스타일 프리셋
+
+  todoItem.innerHTML = `<button class="itemFinish" type="button"><p class="itemTitle">${itemText}</p></button>
+  <button class="itemDelete" type="button"><p><i class="fa fa-solid fa-trash"></i></p></button>`;
+
+  const moveBtn = todoItem.querySelector(".itemFinish");
+  moveBtn.addEventListener("click", moveToDone);
+  const deleteBtn = todoItem.querySelector(".itemDelete");
+  deleteBtn.addEventListener("click", deleteTodoItem);
+  todoList.appendChild(todoItem);
+
+  //각각 moveTodone(doneList로 이동), deleteTodoItem(todoList에서 삭제) 함수 연결
+}
+
+function createDoneItem(id, itemText) {
+  const doneItem = document.createElement("div");
+  let attribute = document.createAttribute("id");
+  attribute.value = id;
+  doneItem.setAttributeNode(attribute);
+  doneItem.classList.add("item");
+
+  doneItem.innerHTML = `<button class="itemFinish" type="button"><p class="itemTitle"><del>${itemText}</del></p></button>
+  <button class="itemDelete" type="button"><p><i class="fa fa-solid fa-trash"></i></p></button>`;
+
+  const moveBtn = doneItem.querySelector(".itemFinish");
+  moveBtn.addEventListener("click", moveToTodo);
+  const deleteBtn = doneItem.querySelector(".itemDelete");
+  deleteBtn.addEventListener("click", deleteDoneItem);
+  doneList.appendChild(doneItem);
+
+  //각각 moveTodone(todoList로 이동), deleteTodoItem(doneList에서 삭제) 함수 연결
+}
+
+// ====== Storage 조작 관련 함수들 ======= //
+// 과거 수강했었던 강의의 코드를 참조했습니다
+// https://www.youtube.com/c/CodingAddict/videos
+//list | {id, {value: value, type: type}} 구조
+
+function getLocalStorage() {
+  return localStorage.getItem("list")
+    ? JSON.parse(localStorage.getItem("list"))
+    : [];
+}
+
+function addToLocalStorage(id, value) {
+  console.log(value);
+  const item = { id, value };
+  let items = getLocalStorage();
+  items.push(item);
```

Comment:
ES6 문법인 `spread` 연산자를 사용해 봅시다! 

```suggestion
  let items = [...getLocalStorage(), item];
```

또한 JS의 `push()` 메소드는 원본 배열의 내용 자체를 직접 변경합니다. 하지만 원본 배열을 직접 변경하는 경우 의도치 않은 결과가 발생할 수도 있기 때문에, 위 코드처럼 새로운 배열로 만드는 방법을 추천드립니다. [이 글](https://songseungwon.tistory.com/entry/JavaScript-Immutability-Push-vs-Concat) 참고하시면 좋을 것 같습니다!

Reply:
감사합니다. spread operator에게 이런 장점이 있다는 것은 처음 알았어요.